Source/EasyNetQ/AdvancedBusExtensions.cs

```diff
@@ -732,7 +732,7 @@ CancellationToken cancellationToken
         /// <param name="routingKey">The routing key</param>
         /// <param name="cancellationToken">The cancellation token</param>
         /// <returns>A binding</returns>
-        public static IBinding Bind(this IAdvancedBus bus, IExchange source, IExchange destination, string routingKey, CancellationToken cancellationToken = default)
+        public static Binding<Exchange> Bind(this IAdvancedBus bus, Exchange source, Exchange destination, string routingKey, CancellationToken cancellationToken = default)
```

Comment:
Consider to use `in` everywhere for `Exchange`, `Queue` and `Binding` arguments. This will make it clear that protective struct copies are not used.

Reply:
As I remember it only works for sync methods

Reply:
So I don't know whether it is worth to use it somewhere in the code.

Reply:
OK